controllers/upgrade_handlers.go

```diff
@@ -53,6 +58,10 @@ func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lc
 		return ctrl.Result{}, err
 	}
 
+	// Save the IBU CR to the new state root before pivot
+	if err := saveIBU(ibu, stateRootRepo); err != nil {
```

Comment:
Why we should have ibu to be different than any other manifest?
Please add this logic to clusterconfig.go as it is part of it

Reply:
clusterconfig is called when we enter the upgrade stage. This one is meant to be done right before the pivot.

Reply:
Also, we are saving the CR that the current reconcile call is working on. There may be status changes comparing to what you would get from a get API call.